app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/login/LoginScreen.kt

```diff
@@ -35,56 +58,165 @@ fun LoginScreen(
+                            text = viewModel.setText(loginField, AuthScreenState.Login),
+                            loginScreenEnum = loginField,
+                            isValid = viewModel.validateEditText(loginField, AuthScreenState.Login),
+                            onValueChange = { newValue ->
+                                viewModel.onEvent(loginField, newValue, AuthScreenState.Login)
+                            },
+                            isLastEditText = index == loginFields.size - 1,
+                            onNextFocusRequest = if (index < focusRequestList.size -1) focusRequestList[index + 1] else focusRequestList[index]
+                        )
+                    }
+                    Text(
+                        modifier = Modifier
+                            .padding(horizontal = 24.dp)
+                            .graphicsLayer {
+                                alpha = if (viewModel.signInData.authError) 1f else 0f
+                            },
+                        text = Constants.AUTHENTICATION_ERROR_TEXT,
+                        color = Color.Red,
+                        style = MaterialTheme.typography.labelMedium,
+                        textAlign = TextAlign.Center
+                    )
+                    LoginScreenButton(
+                        modifier = Modifier
+                            .padding(top = 12.dp)
+                            .padding(horizontal = 24.dp),
+                        isEnabled = viewModel.signInData.isValidLoginPage,
+                        text = LOGIN_TEXT
+                    ) {
+                        scope.launch {
+                            keyboardController?.hide()
+                            viewModel.signInEmailAndPassword(isLoginSucceed)
+                        }
+                    }
+                    Text(
+                        modifier = Modifier
+                            .padding(top = 12.dp)
+                            .clickable { onRegisterClicked() },
+                        text = Constants.REGISTER_TEXT,
+                        color = Color.White,
+                        style = MaterialTheme.typography.labelLarge
+                    )
+                }
+            }
+            item {
+                Text(
+                    modifier = Modifier
+                        .padding(bottom = 16.dp)
+                        .clickable { viewModel.setForgotDialogVisibility(true) },
+                    text = "Forgot Password",
+                    color = Color.White,
+                    style = MaterialTheme.typography.labelLarge
+                )
+            }
         }
-        item {
-            LoginScreenButton(
-                modifier = modifier.padding(top = 12.dp),
-                isEnabled = viewModel.signInData.isValidLoginPage,
-                text = LOGIN_TEXT
+    }
+    ForgotPasswordDialog(viewModel)
+}
+
+@Composable
+fun ForgotPasswordDialog(
```

Comment:
I think it would be better to create a package for dialogs and move the composable function ForgotPasswordDialog into it.